script.js

```diff
@@ -1,4 +1,206 @@
+  return cardDeck;
+};
+
+var getRandomIndex = function (size) {
+  return Math.floor(Math.random() * size);
+};
+// Deck is shuffled.
+/**
+ * Shuffle elements in the cardDeck array. Return the shuffled deck.
+ */
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex to shuffle the next pair of cards
+    currentIndex += 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+
+// Initialise the shuffled card deck before the game starts.
+var deck = shuffleCards(makeDeck());
+
+var blackjackLimit = 21;
+
+var playerHand = [];
+var computerHand = [];
+
+// User clicks Submit to deal cards.
+var dealCard = function (){
+  playerHand.push(deck.pop());
+  computerHand.push(deck.pop());
+  playerHand.push(deck.pop());
+  computerHand.push(deck.pop());
+};
+
+// The cards are analysed for game winning conditions, e.g. Blackjack.
+
+var getHandSum = function(abstractHand) {
+  var numAcesInHand = 0;
+  var sum = 0;
+  var i = 0
+  while (i<abstractHand.length){
+    var currentCard = abstractHand[i];
+    if (currentCard.rank >=2 && currentCard.rank <=10){
+      sum = sum + currentCard.rank;
+    } else if (currentCard.rank >=11 && currentCard.rank <=13){
+      sum = sum + 10;
+    } else if (currentCard.rank == 1) {
+      numAcesInHand = numAcesInHand + 1
+      sum = sum + 11
+    };
+     i +=1;
+     console.log(abstractHand)
+  };
+  if (sum > blackjackLimit && numAcesInHand > 0){
+  i = 0
+  while (i < numAcesInHand){
+    sum = sum - 10
+    i +=1
+  if (sum <= blackjackLimit) {
+  break;
+};
+};
+};
+console.log(sum);
+return sum;
+};
+// The user decides whether to hit or stand, using the submit button to submit their choice.
+// The user's cards are analysed for winning or losing conditions.
+// The computer decides to hit or stand automatically based on game rules.
+// The game either ends or continues.
+var MODE_DEAL_CARDS = "deal and display player cards and 1 computer card";
+var MODE_HIT_OR_STAND = "get another card or stay"
+var MODE_COMPARE_CARD_SUM = "compare sum";
+var mode = MODE_DEAL_CARDS;
+
+var printCardsInHand = function(abstractHand){
+  var cardsInHandString = '';
+  for(var i=0;i<abstractHand.length;i+=1){
+  cardsInHandString += `[${abstractHand[i].name} of ${abstractHand[i].suit}] `;
+  }
+return cardsInHandString
+};
+
+var playerHandSum = 0
+var computerHandSum = 0
+var computerStandLimit = 17
+
 var main = function (input) {
```

Comment:
I like that `main` is relatively short, with code meaningfully abstracted into helper functions :)